ConsoleApp1/DataAccessLayer/SQLAcess/SQLProductAccess.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System.Data.SqlClient;
+using System.Data.SqlTypes;
+
+namespace ConsoleApp1.DataAccessLayer.SQLAcess
+{
+    internal class SQLProductAccess : IProductDataAccess
+    {
+
+        string connectionString = "Data Source=M7mad;Initial Catalog=Order_Managment;Integrated Security=True;TrustServerCertificate=True";
+
+        public bool AddProduct(Product product)
+        {
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+                string query = "INSERT INTO Products (Name, Price, Quantity) VALUES (@ProductName, @Price, @Quantity)";
+
+                using (SqlCommand command = new SqlCommand(query, connection))
```

Comment:
do you think we will need error handling inside the using statment ?

Reply:
i don't know if there is any other aspect that i am missing  but i think i should check for the connection since if there awas a problem with it  we will not enter  what is inside lines 18 to 28